index.js

```diff
@@ -75,43 +73,39 @@ conventionalRecommendedBump({
 })
 
 function outputChangelog (argv, cb) {
-  console.log(chalk.bold('2.') + ' update changelog (' + chalk.bold(argv.infile) + ')')
-
   createIfMissing(argv)
-
-  var readStream = fs.createReadStream(argv.infile)
-    .on('error', function (err) {
-      console.warn(chalk.yellow(err.message))
-    })
-
+  var header = '# Change Log\n\nAll notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.\n'
+  var oldContent = fs.readFileSync(argv.infile, 'utf-8')
+  // find the position of the last release and remove header:
+  if (oldContent.indexOf('<a name=') !== -1) {
+    oldContent = oldContent.substring(oldContent.indexOf('<a name='))
+  }
+  var content = ''
   var changelogStream = conventionalChangelog({
     preset: argv.preset,
     outputUnreleased: true,
     pkg: {
       path: path.resolve(process.cwd(), './package.json')
     }
   })
-    .on('error', function (err) {
-      console.error(chalk.red(err.message))
-      process.exit(1)
-    })
-  var tmp = tempfile()
+  .on('error', function (err) {
+    console.error(chalk.red(err.message))
+    process.exit(1)
+  })
 
-  changelogStream
-    .pipe(addStream(readStream))
-    .pipe(fs.createWriteStream(tmp))
-    .on('finish', function () {
-      fs.createReadStream(tmp)
-        .pipe(fs.createWriteStream(argv.infile))
-        .on('finish', function () {
-          rimraf.sync(tmp)
-          return cb()
-        })
-    })
+  changelogStream.on('data', function (buffer) {
+    content += buffer.toString()
+  })
+
+  changelogStream.on('end', function () {
+    checkpoint('outputting changes to %s', [argv.infile])
+    fs.writeFileSync(argv.infile, header + '\n' + content + oldContent, 'utf-8')
+    return cb()
+  })
 }
 
 function commit (argv, newVersion, cb) {
-  console.log(chalk.bold('3.') + ' commit ' + chalk.bold('package.json') + ' and ' + chalk.bold(argv.infile))
+  checkpoint('commiting %s and %s', ['package.json', 'argv.infile'])
```

Comment:
Here's the culprit, change `'argv.infile'` to `argv.infile`.

Also, note that when using `--first-release` this checkpoint says it is committing package.json, but package.json is not actually modified (no version bump). Might want to make the checkpoint message and args conditional:

```js
var msg = 'committing %s', args = [argv.infile]
if (!argv.firstRelease) {
  msg += ' and %s'
  args.push('package.json')
}
checkpoint(msg, args)
```

or the less verbose:

```js
checkpoint(argv.firstRelease ? 'commiting %s' : 'commiting %s and %s', argv.firstRelease ? [argv.infile] : ['package.json', argv.infile])
```